test/e2e/network/proxy.go

```diff
@@ -258,9 +262,87 @@ var _ = SIGDescribe("Proxy", func() {
 				framework.Failf(strings.Join(errs, "\n"))
 			}
 		})
+
+		ginkgo.It("proxy connection returns a series of 301 redirections for a pod", func() {
+			ns := f.Namespace.Name
+			httpVerbs := []string{"DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"}
+
+			ginkgo.By("Creating a Pod")
+			_, err := f.ClientSet.CoreV1().Pods(ns).Create(context.TODO(), &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "proxy-pod-target",
+					Labels: map[string]string{
+						"test": "redirect"},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Image: imageutils.GetE2EImage(imageutils.Agnhost),
+						Name:  "agnhost",
+					}},
+					RestartPolicy: v1.RestartPolicyNever,
+				}}, metav1.CreateOptions{})
+			framework.ExpectNoError(err, "failed to create pod")
+			framework.Logf("pod created")
+
+			err = wait.PollImmediate(podRetryPeriod, podRetryTimeout, checkPodStatus(f, "test=redirect"))
+			framework.ExpectNoError(err, "Pod didn't start within time out period")
+
+			transportCfg, err := f.ClientConfig().TransportConfig()
+			framework.ExpectNoError(err, "Error accessing TransportConfig")
+			tlsCfg, err := transport.TLSConfigFor(transportCfg)
+			framework.ExpectNoError(err, "Error accessing TLSConfigFor(transportCfg)")
+			// Disable 301 automatic follow
+			client := &http.Client{
+				CheckRedirect: func(req *http.Request, via []*http.Request) error {
+					return http.ErrUseLastResponse
+				},
+				Transport: &http.Transport{TLSClientConfig: tlsCfg},
```

Comment:
this can probably simplify to something like `Transport: transport.New(f.ClientConfig())`, and drop the TLS config construction and manual bearer token header construction

Reply:
I believe the issues listed above have been removed/addressed now. Happy to updated it further if you have any other feedback/tips.